plugin-infra/go-plugin-infra/src/com/thoughtworks/go/plugin/infra/monitor/DefaultPluginJarLocationMonitor.java

```diff
@@ -181,40 +183,50 @@ public void run() {
 
         private Set<PluginFileDetails> loadAndNotifyPluginsFrom(File pluginDirectory, Set<PluginFileDetails> knownPluginFiles, boolean isBundledPluginsLocation) {
             Set<PluginFileDetails> currentPluginFiles = getDetailsOfCurrentPluginFilesFrom(pluginDirectory, isBundledPluginsLocation);
-            notifyListenersOfRemovedPlugins(currentPluginFiles, knownPluginFiles);
-            notifyListenersOfUpdatedPlugins(currentPluginFiles, knownPluginFiles);
-            notifyListenersOfAddedPlugins(currentPluginFiles, knownPluginFiles);
+            boolean notifyListenersOfRemovedPlugins = notifyListenersOfRemovedPlugins(currentPluginFiles, knownPluginFiles);
+            boolean notifyListenersOfUpdatedPlugins = notifyListenersOfUpdatedPlugins(currentPluginFiles, knownPluginFiles);
+            boolean notifyListenersOfAddedPlugins = notifyListenersOfAddedPlugins(currentPluginFiles, knownPluginFiles);
+            if (notifyListenersOfRemovedPlugins || notifyListenersOfUpdatedPlugins || notifyListenersOfAddedPlugins) {
```

Comment:
can this be made more simpler? computing 3 booleans & checking looks a little clumpsy.